spring-cloud-gcp-starters/spring-cloud-gcp-starter-logging/src/main/java/org/springframework/cloud/gcp/logging/autoconfig/TraceIdExtractorType.java

```diff
@@ -0,0 +1,32 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.logging.autoconfig;
+
+/**
+ * Enum values represent the order and combination of predefined trace ID extractors that
+ * can be used.
+ *
+ * @see XCloudTraceIdExtractor
+ * @see ZipkinTraceIdExtractor
+ * @see CompositeTraceIdExtractor
+ * @see TraceIdExtractor
+ *
+ * @author Chengyuan Zhao
+ */
+public enum TraceIdExtractorType {
+	XCLOUD, ZIPKIN, XCLOUD_ZIPKIN, ZIPKIN_XCLOUD
```

Comment:
You can include more documentation for each of the enum values.
See: https://stackoverflow.com/questions/6549821/how-to-javadoc-a-classs-individual-enums.

Reply:
We have this other issue for all the documentation and I was going to make all those  changes then: #288 . I didn't want to make the readme changes in this PR because as you see folks are switching it up and I didn't want to rewrite the readme every round thinking it might be final.

Reply:
#288 is specific to [Reference Documentation](https://docs.spring.io/spring-cloud-gcp/docs/1.0.0.BUILD-SNAPSHOT/reference/htmlsingle/) in which we have no mention of Logging yet at all.

I'm talking about README doc updates just for this particular change of introducing the extractor option.

Reply:
Ah, ok I will do that in a following PR.

Reply:
https://github.com/spring-cloud/spring-cloud-gcp/issues/310

Reply:
https://github.com/spring-cloud/spring-cloud-gcp/issues/310